libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -140,7 +141,18 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
     @Test
     fun init_routesObs_internalRouteObs_navigationSession_and_TelemetryLocAndProgressDispatcher() {
         createMapboxNavigation()
-        verify(exactly = 2) { directionsSession.registerRoutesObserver(any()) }
+        // + 1 for navigationSession
+        // + 1 for routesCacheClearer
+        verify(exactly = 3) { directionsSession.registerRoutesObserver(any()) }
```

Comment:
I'm wondering, why do we need to test this implementation detail at all 🤔

Reply:
It would make more sense to capture the argument and check the observer by its actions. I think for internalRoutesObserver it's tested somewhere.